Feat/coupon : 쿠폰 수정 기능 생성 및 에러 컨트롤러 추가, 테스트 코드 작성

gloz0315 opened the pull request:
## 변경사항

- 쿠폰 수정 기능 생성
- 에러 컨트롤러 추가
- 쿠폰 생성 및 수정시 날짜에 대한 처리기능 추가
- 테스트 코드 작성 및 성공

### 변경의 종류 (여러 가지 선택 가능)

- [x] 새로운 기능
- [ ] 버그 수정
- [ ] 문서 업데이트
- [x] 테스트 추가
- [x] 기타(설명 추가)

---

## 체크리스트

- [x] ERD설계와 API설계를 잘 적용하여 개발하였는가?
- [x] 서비스 테스트의 커버리지가 80% 이상인가?
- [x] 코딩 컨벤션을 준수하였는가?
- [x] 예외 처리가 잘 되어 있는가?
- [x] 커밋 메시지를 컨벤션에 맞게 작성 하였는가?

gloz0315 committed:
✨ feat(coupon): 쿠폰 수정 기능 생성

gloz0315 committed:
✨ feat(coupon): 쿠폰 DTO 클래스 생성

gloz0315 committed:
✨ feat(coupon): 쿠폰 수정 or 생성시 시작일과 마감일에 대한 검사 기능 추가

- 에러 핸들러 추가
- Custom 에러 클래스 생성

gloz0315 committed:
✅ test(coupon): 쿠폰 수정에 관한 테스트 코드 작성

tktcod commented on src/main/java/com/junwoo/ott/domain/coupon/dto/body/CouponCreateDto.java:
```diff
@@ -1,30 +1,41 @@
+
+  public CouponCreateDto(String description, CouponType type, MembershipType membershipType,
+      Integer discount, Integer count, String startAt, String endAt) {
+    Coupon.validateDate(startAt, endAt);
```
@NotNull이 있는데도 validate에서도 null 검사하고 있는거 같습니다
저희 controller에서 받는 dto와 create가 같아도 새로운 dto 만드는 것이 아니었나요?

gloz0315 replied:
네 CreateReqeustDto는 따로 있습니다.
다시 확인을 해 보니 CreateDto에는 @NotNull 검사가 되어 있어서 따로 null처리 관련된 처리를 할 필요가 없다 생각하지만 validateDate 메서드 안에 보면 startAt과 endAt의 기간에 대한 예외처리 방식을 사용하고 있어서 적용했습니다.
이 부분은 다시 메서드를 분리하여 수정했습니다.

gloz0315 committed:
🎨 style(coupon): validateDate 메서드 수정

- CreateDto에는 @NotNull 어노테이션으로 날짜에 대한 null 로직이 불필요 -> 수정

- UpdateDto에는 날짜에 대한 null 로직이 필요하기 때문에 이 부분에 대해 메서드를 분리

gloz0315 committed:
🎨 style(coupon): 줄바꿈 변경

ersesd reviewed (approved):
LGTM

devJWL reviewed (approved):
LGTM

ruh0n reviewed (approved):
LGTM

gloz0315 committed:
🎨 style(coupon): 파라미터에 final 붙이기 수정

gloz0315 committed:
🎨 style(coupon): @Secured 어노테이션 적용 및 파라미터 final 적용

- 테스트 코드에 CreateRequestDto에 더이상 id값이 필요 없기에 수정
